File: Courses/Models/Venue.cs
using System.ComponentModel.DataAnnotations;

namespace Courses.Models;

public class Venue : IHasTimestamp
{
    public Guid Id { get; set; } = Guid.CreateVersion7();

    [StringLength(255)] public required string NameEn { get; set; }
    [StringLength(255)] public string? NamePl { get; set; }
    [StringLength(255)] public required string Street { get; set; }
    [StringLength(255)] public required string CityEn { get; set; }
    [StringLength(255)] public string? CityPl { get; set; }

    [StringLength(15)] public string? PostalCode { get; set; }

    [StringLength(2)] public required string CountryCode { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}

File: CoursesWeb/src/api/proto/courses/Venue.ts
// Original file: ../Courses/Protos/courses.proto

import type { UUID as _courses_UUID, UUID__Output as _courses_UUID__Output } from '../courses/UUID';
import type { Timestamp as _google_protobuf_Timestamp, Timestamp__Output as _google_protobuf_Timestamp__Output } from '../google/protobuf/Timestamp';

export interface Venue {
  'id'?: (_courses_UUID | null);
  'nameEn'?: (string);
  'namePl'?: (string);
  'street'?: (string);
  'cityEn'?: (string);
  'cityPl'?: (string);
  'postalCode'?: (string);
  'countryCode'?: (string);
  'createdAt'?: (_google_protobuf_Timestamp | null);
  'updatedAt'?: (_google_protobuf_Timestamp | null);
  '_namePl'?: "namePl";
  '_cityPl'?: "cityPl";
  '_postalCode'?: "postalCode";
}

export interface Venue__Output {
  'id': (_courses_UUID__Output | null);
  'nameEn': (string);
  'namePl'?: (string);
  'street': (string);
  'cityEn': (string);
  'cityPl'?: (string);
  'postalCode'?: (string);
  'countryCode': (string);
  'createdAt': (_google_protobuf_Timestamp__Output | null);
  'updatedAt': (_google_protobuf_Timestamp__Output | null);
  '_namePl'?: "namePl";
  '_cityPl'?: "cityPl";
  '_postalCode'?: "postalCode";
}
